lib/api_resources/root.js

```diff
@@ -27,29 +28,119 @@ RootResource.prototype.proxy = function(env, next) {
 
-    if(this.shouldProxy(server)) {
+    if (server === '*') {
+      this.server.peerRegistry.find(Query.of('peers'), function(err, peers) {
+        if (err) {
+          env.response.statusCode = 500;
+          return next(env);
+        }
+
+        var href = '/servers/{{peerName}}';
+        var qs ='?' + querystring.stringify({ql: params.ql});
+        env.request.templateUrl = href + qs;
+
+        var httpServer = self.server.httpServer;
+
+        httpServer.proxyToPeers(peers, env, function(err, results) {
+          var messageId;
+
+          var entities = results.map(function(obj) {
+            if (obj.err) {
+              return [];
+            }
+
+            var response = obj.res;
+            var body = obj.body;
+
+            var id = response.headers['zetta-message-id'];
+
+            if (!messageId) {
+              messageId = id;
+            }
+
+            var res = httpServer.clients[id];
+
+            if (!res) {
+              return [];
+            }
+
+            Object.keys(response.headers).forEach(function(header) {
+              if (header !== 'zetta-message-id') {
+                res.setHeader(header, response.headers[header]);
+              }
+            });
+
+            res.statusCode = response.statusCode;
+
+            if (body) {
+              body = JSON.parse(body.toString());
+              return body && body.entities ? body.entities : [];
+            } else {
+              return [];
+            }
+          }).reduce(function(prev, curr) {
+            return prev.concat(curr);
+          }, []);
+
+          self.server.runtime.registry.find(params.ql, function(err, results) {
+            var localEntities = (!!err || !results) ? [] : results.map(function(device){
+              var deviceOnRuntime = self.server.runtime._jsDevices[device.id];
+              if (deviceOnRuntime) {
+                var model = deviceOnRuntime;
+                var loader = {path: '/servers/' + encodeURI(self.server.id)}
+                var server = self.server;
+                var entity = {
+                  class: ['device'],
+                  rel: [rel.device],
+                  properties: model.properties(),
+                  links: [{ rel: ['self'], href: env.helpers.url.path(loader.path + '/devices/' + model.id) },
+                          { title: server._name, rel: ['up', rel.server], href: env.helpers.url.path(loader.path) }]
+                };
+                return entity;
+              }
+            }).filter(function(entity) { return entity !== null });
+
+            var res = httpServer.clients[messageId];
+            var obj = {
+              class: ['root', 'search-results'],
+              properties: {
+                ql: params.ql
+              },
+              entities: localEntities.concat(entities),
+              links: [
+                { rel: ['self'], href: env.helpers.url.current() },
+                { rel: [rel.query], href: '' }
```

Comment:
Latest commit removes blank query relation and includes `"server": "*"` in
the properties.

On Fri, Apr 24, 2015 at 10:49 AM, Adam Magaluk <notifications@github.com>
wrote:

> In lib/api_resources/root.js
> <https://github.com/zettajs/zetta/pull/178#discussion_r29054711>:
>
> > +                          { title: server._name, rel: ['up', rel.server], href: env.helpers.url.path(loader.path) }]
> > +                };
> > +                return entity;
> > +              }
> > +            }).filter(function(entity) { return entity !== null });
> > +
> > +            var res = httpServer.clients[messageId];
> > +            var obj = {
> > +              class: ['root', 'search-results'],
> > +              properties: {
> > +                ql: params.ql
> > +              },
> > +              entities: localEntities.concat(entities),
> > +              links: [
> > +                { rel: ['self'], href: env.helpers.url.current() },
> > +                { rel: [rel.query], href: '' }
>
> It looks like just gotta remove this line for @mdobson
> <https://github.com/mdobson> reported issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zettajs/zetta/pull/178/files#r29054711>.
>



-- 
Kevin Swiber
Projects: https://github.com/kevinswiber
Twitter: @kevinswiber